samples/Samples.SqlServer/Program.cs

```diff
@@ -25,12 +25,25 @@ private static async Task Main()
 
         private static SqlConnection OpenConnection()
         {
+            int numAttempts = 3;
             var connectionString = Environment.GetEnvironmentVariable("SQLSERVER_CONNECTION_STRING") ??
-                                   @"Server=(localdb)\MSSQLLocalDB;Integrated Security=true;Connection Timeout=30";
+@"Server=(localdb)\MSSQLLocalDB;Integrated Security=true;Connection Timeout=30";
 
-            var connection = new SqlConnection(connectionString);
-            connection.Open();
-            return connection;
+            for (int i = 0; i < numAttempts; i++)
+            {
+                try
+                {
+                    var connection = new SqlConnection(connectionString);
+                    connection.Open();
+                    return connection;
+                }
+                catch (Exception ex)
```

Comment:
Since we do multiple tries, we should dispose the connection in the failure case. Better safe than sorry

Reply:
Fixed in 4bc6e8c440ce77eb56e7217e6aaaaa5e6593060b

Reply:
Nitpicking: in theory, the constructor could throw an exception, so it should be wrapped in the try/catch block:

```csharp
SqlConnection connection = null;

try
{
    connection = new SqlConnection(connectionString);
    ...
catch (Exception ex)
{
    connection?.Dispose();
}

```

BUT exceptions thrown by the constructor are not transient ones, so it doesn't really matter

Reply:
Yeah that makes sense, I'll go ahead and do that since that's really the best thing to do, I shouldn't be sloppy 😄

Reply:
Done in ab7e277381c8791cf87bf914150014eb2ddf57d8